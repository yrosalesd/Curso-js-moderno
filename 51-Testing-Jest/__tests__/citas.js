import Citas from '../js/classes/Citas'; 

describe('Probar la clase de Citas', () => {

    const citas=new Citas(); //instanciando 
    const id=Date.now(); //global para que sepa que id es, y sea el mismo id para ambas pruebas  

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Laika',
            propietario: 'Julieta',
            telefono: '123456',
            fecha: '10-12-2020',
            hora:'10:30',
            sintomas: 'solo duerme'
        }; //parametros que le paso estan aqui porque no los importamos 

        

        citas.agregarCita(citaObj); // se usa la instanciada 

        //prueba
        expect(citas).toMatchSnapshot();
        //               ---------------------------verificar que info que estamos pasando a la clase, se este guardando correctamente, escribe un snapshot con el arreglo y la info 
        
    }); 

    test('actualizar cita', () => {
        const citaActualizada={
            id,
            mascota: 'Nuevo Nombre',
            propietario:'Julieta',
            telefono:'19999393',
            fecha:'10-12-2020',
            hora:'10:30',
            sintomas:'Solo duerme'
        }; 

        citas.editarCita(citaActualizada); 

        expect(citas).toMatchSnapshot(); // es correcto porque esta actulizada basado en el id 
    }); 

    test('Eliminar Cita', () => {
         citas.eliminarCita(id); 

         expect(citas).toMatchSnapshot(); 
    })
}); 

