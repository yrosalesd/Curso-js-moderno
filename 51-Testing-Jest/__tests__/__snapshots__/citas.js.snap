// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1616102693133,
      "mascota": "Laika",
      "propietario": "Julieta",
      "sintomas": "solo duerme",
      "telefono": "123456",
    },
  ],
}
`;

exports[`Probar la clase de Citas Eliminar Cita 1`] = `
Citas {
  "citas": Array [],
}
`;

exports[`Probar la clase de Citas actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2020",
      "hora": "10:30",
      "id": 1616102693133,
      "mascota": "Nuevo Nombre",
      "propietario": "Julieta",
      "sintomas": "Solo duerme",
      "telefono": "19999393",
    },
  ],
}
`;
